### PR TITLE
Add interstitial for Stats

### PR DESCRIPTION
**Pull request by jboland88**
## Proposed changes:

* This PR adds a product interstitial for Stats in My Jetpack

image

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pbNhbs-9bI-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Spin up a JN site with Jetpack Beta pointing to this branch
* Visit `/wp-admin/admin.php?page=my-jetpack#/add-stats`
* Confirm the new interstitial matches the screenshot above, and the CTA works
* Test the CTA on the My Jetpack stats card - without a plan it should say "learn more". With a free or tiered stats plan, it should say "Upgrade". With a PWYW or legacy commercial plan, it should say "View"

### CODE REVIEW

File: projects/packages/my-jetpack/src/products/class-stats.php [hunk 1]
```diff
@@ -162,23 +173,21 @@ public static function is_upgradable() {
...
 				}
-			}
+			);
+
+			return ! empty( $upgradeable_stats_purchases );
 		}
-		return true;
+
+		// If there are no plans found, don't consider the product as upgradeable
+		return false;
```
Comment (CodeyGuyDylan) [inline projects/packages/my-jetpack/src/products/class-stats.php:205]: Just curious, what do you think the reasoning behind defaulting to `true` was before? I can't think of one, just want to make sure there isn't a good reason we are missing

**Review by CodeyGuyDylan — DISMISSED** [review]
This LGTM, just one question 😄

File: projects/packages/my-jetpack/src/products/class-stats.php (hunk 1 above)
Comment (jboland88) [reply projects/packages/my-jetpack/src/products/class-stats.php:205]: Good question -  I am not quite sure either 🤔 
It seems like the old approach was a blacklist approach and the new approach is more of a whitelist 🤷

**Review by CodeyGuyDylan — DISMISSED** [review]
When I have a Stats commercial license, I see the "Upgrade" button but am taken to Stats PWYW when I click on it, which would be a downgrade
image
image

**Comment by robertsreberski** [comment]
After the change, clicking update when having Commercial license leads to Commercial PWYW:
CleanShot 2024-01-08 at 12 33 38@2x

LGTM

**Review by robertsreberski — APPROVED** [review]

**Pull request merged**